crates/bevy_animation/src/animation_curves.rs

```diff
@@ -205,8 +274,7 @@ where
     P: AnimatableProperty,
 {
     evaluator: BasicAnimationCurveEvaluator<P::Property>,
-    #[reflect(ignore)]
-    phantom: PhantomData<P>,
+    property: P,
```

Comment:
This is basically my only concern with this PR. 

Since evaluators are cached and dispatched based on type IDs, the previous setup with marker structs guaranteed that all `AnimatableCurve`s accessing the same property would map to evaluators with the same type, since they used the same marker type as input. 

Now, since `AnimatedProperty` has a function-type generic `F: Fn(&mut C) -> &mut P`, I think there is a subtle footgun involving the instantiation of evaluators: for example, if you create two `AnimatableCurve`s using `AnimatableProperty`s that are the same, but use closures to specify the property accessor, the two will map to distinct evaluator types, since the types (`P` here) will actually be distinct. As a result, the outputs of those curves will not blend correctly (one will just overwrite the other, I think).

In other words, the previous implementation effectively guaranteed that the property accessors would be type-identical, and that is no longer the case now.

Reply:
(It might be enough to just warn about this and recommend defining the property in a large lexical scope and then reusing it in calls to `AnimatableCurve::new`; if anything, I'm mostly pointing out that the pattern described in the 'Solution' section of the PR description is subtly an anti-pattern.)

Reply:
Hmm yeah nice catch. Current behavior doesn't feel shippable, but I think it points to a more fundamental problem.  Theres an odd coupling right now between evaluators and "property accessor code".

Seems like we should decouple evaluator identity from a statically defined AnimationProperty type, and instead tie it to the id of the component combined with the id of the property/field being accessed. I think we'll want to do that anyway to support things like `ReflectedAnimatableProperty::new(TypeId::of::<Transform>(), "translation")`.

Reply:
I'll start playing with that.

Reply:
This would probably also allow making arbitrary transform field animation (via things like `AnimatableProperty`) compatible with a custom `TransformCurve` impl. Rather than using the #15665 approach, which relies on everything using the same custom impls.